support/cas-server-support-aup-ldap/src/test/java/org/apereo/cas/aup/LdapMissingAcceptableUsagePolicyRepositoryTests.java

```diff
@@ -0,0 +1,87 @@
+package org.apereo.cas.aup;
+
+import com.unboundid.ldap.sdk.LDAPConnection;
+import lombok.Cleanup;
+import lombok.Getter;
+import lombok.val;
+import org.apache.commons.io.IOUtils;
+import org.apereo.cas.adaptors.ldap.LdapIntegrationTestsOperations;
+import org.apereo.cas.config.CasAcceptableUsagePolicyLdapConfiguration;
+import org.apereo.cas.util.CollectionUtils;
+import org.apereo.cas.util.RandomUtils;
+import org.apereo.cas.util.junit.EnabledIfPortOpen;
+import org.apereo.inspektr.common.web.ClientInfo;
+import org.apereo.inspektr.common.web.ClientInfoHolder;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.test.context.TestPropertySource;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * This is {@link LdapMissingAcceptableUsagePolicyRepositoryTests}.
+ *
+ * @author Miguel Martinez de Espronceda
+ * @since 6.5.0
+ */
+@Tag("Ldap")
+@EnabledIfPortOpen(port = 10389)
+@Import(CasAcceptableUsagePolicyLdapConfiguration.class)
+@TestPropertySource(properties = {
+    "cas.acceptable-usage-policy.ldap[0].ldap-url=ldap://localhost:10389",
+    "cas.acceptable-usage-policy.ldap[0].base-dn=ou=people,dc=example,dc=org",
+    "cas.acceptable-usage-policy.ldap[0].search-filter=cn={0}",
+    "cas.acceptable-usage-policy.ldap[0].bind-dn=cn=Directory Manager",
+    "cas.acceptable-usage-policy.ldap[0].bind-credential=password",
+    "cas.acceptable-usage-policy.core.aup-attribute-name=carLicense",
+    "cas.acceptable-usage-policy.core.aup-omit-if-attribute-missing=true"
+})
+@Getter
+public class LdapMissingAcceptableUsagePolicyRepositoryTests extends BaseAcceptableUsagePolicyRepositoryTests {
```

Comment:
Remove this test class please and do not duplicate an entire class for a test. Instead, add your test case to an existing relevant class.

Reply:
Hi mmoayyed! I added an additional test just because it has a property to be defined (cas.acceptable-usage-policy.core.aup-omit-if-attribute-missing=true). Which will be the correct way? Thank you!

Reply:
If this setting cannot be added to an existing class and interferes with current test behavior, you need to use `@Nested` tests. 

See `SendTicketGrantingTicketActionTests` for an example please.

Reply:
Good! Did not know that annotation. I have remade it as @Nested.